packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -116,12 +116,12 @@ class HotRunner extends ResidentRunner {
         final CompilerOutput compilerOutput =
             await device.generator.compileExpression(expression, definitions,
                 typeDefinitions, libraryUri, klass, isStatic);
-        if (compilerOutput.outputFilename != null) {
+        if (compilerOutput != null && compilerOutput.outputFilename != null) {
```

Comment:
Under what circumstances do we end up with a null compilerOutput?

Reply:
That could happen if compiler was not started or if compiler process crashed unexpectedly.